devtools/etdump/data_sink_base.h

```diff
@@ -0,0 +1,49 @@
+// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+
+#pragma once
+
+#include <executorch/runtime/core/exec_aten/exec_aten.h>
+#include <cstdlib>
+
+namespace executorch {
+namespace etdump {
+
+/**
+ * DataSinkBase is an abstract class that users can inherit and implement
+ * to customize the storage and management of debug data in ETDumpGen. This
+ * class provides an basic and essential interface for writing tensor data to a
+ * user-defined storage, retrieving storage capacity, and tracking the amount of
+ * data stored.
+ */
+class DataSinkBase {
+ public:
+  /**
+   * Virtual destructor to ensure proper cleanup of derived classes.
+   */
+  virtual ~DataSinkBase() = default;
+  /**
+   * Write tensor data into the debug storage. This method should be implemented
+   * by derived classes to handle the specifics of data storage.
+   *
+   * @param[in] tensor The tensor data to be written into the storage.
+   * @return The offset of the starting location of the tensor data within the
```

Comment:
Returning only the offset isn't enough to be safe: it's be better to also return the length, so that readers of this data know how much they're allowed to read at that offset.

Reply:
i think that should be fine cuz user knows how many data they want to be written in the buffer.

If the write succeeds, the number of bytes should be allowed to read at that offset is just equal to the number of bytes want to write;
If not, the function will return an Error.